src/transformers/models/sam/image_processing_sam.py

```diff
@@ -398,9 +399,16 @@ def post_process_masks(
         requires_backends(self, ["torch"])
         pad_size = self.pad_size if pad_size is None else pad_size
         target_image_size = (pad_size["height"], pad_size["width"])
-
+        if isinstance(original_sizes, (torch.Tensor, np.ndarray)):
+            original_sizes = original_sizes.tolist()
+        if isinstance(reshaped_input_sizes, (torch.Tensor, np.ndarray)):
+            reshaped_input_sizes = reshaped_input_sizes.tolist()
         output_masks = []
         for i, original_size in enumerate(original_sizes):
+            if isinstance(masks[i], np.ndarray):
+                masks[i] = torch.from_numpy(masks[i])
+            elif isinstance(masks[i], List):
+                raise ValueError("Input masks should be a list of `torch.tensors` or a list of `np.ndarray`")
```

Comment:
I would just have `else` here - testing for list is very specific (is there a reason for checking this specific type?) 
```suggestion
            else:
                raise ValueError("Input masks should be a list of `torch.tensors` or a list of `np.ndarray`")
```

Reply:
Hmm I think this seems to be needed, I just added another check for testing if `masks[i]` is a tensor

Reply:
Otherwise the post processing fails